src/assets/scripts/client/aircraft/AircraftModel.js

```diff
@@ -1714,19 +1698,21 @@ export default class AircraftModel {
      * This will sets up and prepares the aircraft to hold
      *
      * @for AircraftModel
-     * @method updateTargetHeadingForHold
-     */
-    updateTargetHeadingForHold() {
-        const invalidTimerValue = -999;
-        const { hold } = this.fms.currentWaypoint;
-        const outboundHeading = radians_normalize(hold.inboundHeading + Math.PI);
-        const offset = getOffset(this, hold.fixPos, hold.inboundHeading);
-        const holdLegDurationInSeconds = hold.legLength * TIME.ONE_MINUTE_IN_SECONDS;
-        const bearingToHoldFix = vradial(vsub(hold.fixPos, this.relativePosition));
+     * @method _calculateTargetedHeadingHold
+     */
+    _calculateTargetedHeadingHold() {
+        const currentWaypoint = this.fms.currentWaypoint;
+        const holdParameters = currentWaypoint.holdParameters;
+        const waypointRelativePosition = currentWaypoint.relativePosition;
+        const outboundHeading = radians_normalize(holdParameters.inboundHeading + Math.PI);
+        const offset = getOffset(this, waypointRelativePosition, holdParameters.inboundHeading);
+        const holdLegDurationInMinutes = holdParameters.legLength.replace('min', '');
+        const holdLegDurationInSeconds = holdLegDurationInMinutes * TIME.ONE_MINUTE_IN_SECONDS;
```

Comment:
why not calculate these two on creation of a waypoint?  that way you can simply reference `holdParameters.legDurationInMinutes` and `holdParameters.legDurationInSeconds`